font-codegen/src/table.rs

```diff
@@ -418,6 +427,12 @@ pub(crate) fn generate_format_compile(
         quote! ( #( #docs )* #name(#typ) )
     });
 
+    let default_variant = &item
+        .variants
+        .first()
+        .expect("empty format groups not supported")
+        .name;
```

Comment:
This feels misplaced to me, shouldn't having >0 variants be an invariant of generic group, checked when parsing and then safe to trust thereafter? - if so you could unwrap because you *know* it's not empty.

I'm actually somewhat surprised the type system doesn't facilitate expressing a "not empty" vector.

Reply:
Also this is duplicative of above, maybe both should be calling the same helper?

Reply:
This is a case of me using `expect` to document why a given unwrap should not be a problem in practice, but I'm happy either way.